### PR TITLE
FIX: amount of player required at creation

### PR DESCRIPTION
**Pull request by Segelzwerg**
# Fixes
- Removes the field `team_size` from `Matcher` class. This field required to know the amount of players at the creation of a table. However, we allow dynamic amount of players if required. So the amount of players at the start of a table could be less than originally planned for. This changes the behavior of matchers to dynamically adapt during the request of a new distribution.

### CODE REVIEW

File: tests/whist_core/session/test_matcher.py [hunk 1]
```diff
@@ -14,8 +14,8 @@ def setUp(self) -> None:
         player_d = Player(user_id=5, username='d', rating=1)
         self.players = [player_a, player_b, player_c, player_d]
         self.user_list = UserList()
-        self.random_matcher = RandomMatcher(number_teams=2, team_size=2)
-        self.robin_matcher = RoundRobinMatcher(number_teams=2, team_size=2)
+        self.random_matcher = RandomMatcher(number_teams=2)
```
[the remark this answers is not in the document]
Comment (Segelzwerg) [reply tests/whist_core/session/test_matcher.py:17]: False positive: pydantic

File: pyproject.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [tool.poetry]
 name = "whist-core"
 # remember to also update the version in __init__.py!
-version = "0.7.0"
+version = "0.7.1"
```
Comment (iTitus) [inline pyproject.toml:4]: why the version increase already?

File: whist_core/__init__.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 import os
 
 # remember to also update the version in pyproject.toml!
-__version__ = '0.7.0'
+__version__ = '0.7.1'
```
Comment (iTitus) [inline whist_core/__init__.py:7]: same here

File: pyproject.toml (hunk 2 above)
Comment (Segelzwerg) [reply pyproject.toml:4]: This fix is required for next steps in server so we need to release it 😉

**Review by iTitus — CHANGES_REQUESTED** [review]
technically this is a breaking change, so dont increase the patch version

**Review by iTitus — APPROVED** [review]

**Pull request merged**